spring-cloud-consul-config/src/main/java/org/springframework/cloud/consul/config/ConsulConfigWatch.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2013-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.consul.config;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.bootstrap.BootstrapApplicationListener;
+import org.springframework.cloud.context.environment.EnvironmentChangeEvent;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationEventPublisherAware;
+import org.springframework.core.env.CompositePropertySource;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.PropertySource;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import com.ecwid.consul.v1.QueryParams;
+import com.ecwid.consul.v1.Response;
+import com.ecwid.consul.v1.kv.model.GetValue;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * A kv store watch that publishes an EnvironmentChangeEvent whenever a value
+ * under one of the property sources is changed
+ * @author Andrew DePompa
+ */
+@Slf4j
+public class ConsulConfigWatch implements ApplicationEventPublisherAware {
+
+    private ApplicationEventPublisher publisher;
+
+    @Autowired
+    private ConfigurableEnvironment environment;
```

Comment:
In practice we only use `@Autowired` in `@Configuration` classes.  Please change this to a constructor arg or implement `EnvironmentAware`.